stellar-multitenant/pkg/serve/serve.go

```diff
@@ -113,16 +115,21 @@ func handleHTTP(opts *ServeOptions) *chi.Mux {
 		Version:   opts.Version,
 	}.ServeHTTP)
 
-	mux.Route("/tenants", func(r chi.Router) {
-		tenantsHandler := httphandler.TenantsHandler{
-			Manager:             opts.tenantManager,
-			ProvisioningManager: opts.tenantProvisioningManager,
-			NetworkType:         opts.networkType,
-		}
-		r.Get("/", tenantsHandler.GetAll)
-		r.Post("/", tenantsHandler.Post)
-		r.Get("/{arg}", tenantsHandler.GetByIDOrName)
-		r.Patch("/{id}", tenantsHandler.Patch)
+	// Authenticated Routes
+	mux.Group(func(r chi.Router) {
+		r.Use(middleware.BasicAuthMiddleware(opts.AdminAccount, opts.AdminApiKey))
```

Comment:
💡  Another test that'd be useful to guarantee we don't remove this authentication by mistake is something similar to what we have below, in the tests `Test_handleHTTP_authenticatedEndpoints` and `Test_handleHTTP_unauthenticatedEndpoints`:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/f8e0abd694b5b9b433d78809d4d15a1a5520ba84/internal/serve/serve_test.go#L310-L421

Reply:
That's great! I'll add that

Reply:
Addressed in 7e46b31ed9f5b608d69cf709f4917bc83214a7ef